### PR TITLE
Implement calculator

### PR DESCRIPTION
**Pull request by SChandarana**

### CODE REVIEW

File: CalculatorTests/Tests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using NUnit.Framework;
+using static Calculator.Program;
+
+namespace CalculatorTests
+{
```
Comment (SChandarana) [inline CalculatorTests/Tests.cs:6]: Might be worth adding tests for what happens if negative integers, floats or strings are given. And also maybe testing combinations of different input types? We'll also need some tests for the other types of operators (-,*,/) and also to check what happens if an invalid operator has been given

File: CalculatorTests/Tests.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            int answer = CalculateAnswer("+", new[] {1, 2, 3, 4, 5, 6});
+            Assert.AreEqual(21, answer);
+        }
+
+        [Test]
+        public void ThrowException()
+        {
+            Assert.Throws<IndexOutOfRangeException>(() => CalculateAnswer("+", new int[] {}));
+        }
```
Comment (SChandarana) [inline CalculatorTests/Tests.cs:35]: I'm not sure if these test names are descriptive enough. I think it would be better to have the test name describe what the expected outcome  e.g `ThrowsExceptionWhenNoNumbersAreGiven` and `CanAddManyNumbers`. What do you think?

File: CalculatorTests/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+//      Major Version
+//      Minor Version 
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers 
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
+[assembly: AssemblyFileVersion("1.0.0.0")]
```
Comment (SChandarana) [inline CalculatorTests/Properties/AssemblyInfo.cs:35]: Make sure files have new lines. Most IDEs have a setting that lets you automatically add new lines to the end of files when they are saved. It might also be worth for us to have a linting step that would check for this, I'll look into this :D

File: Program.cs [hunk 4]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
     {
       Console.WriteLine("Welcome to the calculator!");
       Console.WriteLine("==========================");
+    }
+
+    private static void PerformOneCalculation()
+    {
+      var op = AskForOperator();
+      var n = AskForNumberArray(op);
+      var a = CalculateAnswer(op, n);
```
Comment (SChandarana) [inline Program.cs:27]: These variable names aren't very descriptive making it a little hard to remember exactly what they mean when they're used later (e.g. `n` is often used for a single integer but in this case its a list). I would recommend using full words to describe most variables (unless they are things like `i` in a for loop etc) to make the code clearer for outside readers :D

File: Program.cs [hunk 5]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+      Console.WriteLine("The answer is: " + a);
+      Console.WriteLine();
+    }
+
+    private static string AskForOperator()
+    {
+      string[] allowedOperators = { "+", "-", "*", "/" };
+      Console.Write("Please enter the operator: ");
+      string op = Console.ReadLine();
+      return op;
```
Comment (SChandarana) [inline Program.cs:38]: The `allowedOperators` are never used here. I think we can use it for some input validation to make sure only a valid operator has been used. Once you've done this make sure you add tests for it! :D We can also change this from a string[] to an enum type instead. Does that sound good?

File: Program.cs [hunk 6]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+      for (int index = 0; index < count; index++)
+      {
+        numbers[index] = AddNumber($"Please enter number {index + 1}: ");
+      }
+      return numbers;
+    }
+
+    private static int AddNumber(string message)
+    {
+//      Console.Write(message);
```
Comment (SChandarana) [inline Program.cs:57]: Should this be uncommented?

File: Program.cs [hunk 7]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+
+      int.TryParse(Console.ReadLine(), out number);
+
+      return number;
+    }
+
+    public static int CalculateAnswer(string op, int[] numbers)
+    {
+      // Returns 0 if there aren't any numbers.
+      int answer = numbers[0];
```
Comment (SChandarana) [inline Program.cs:69]: This is actually taking the first element of the input list, meaning that it will throw an error if there are no elements in the list rather than returning 0 as the comment suggests. I would recommend doing something like `if numbers.isEmpty() { return 0 }`
After doing this the comment won't be needed as the code describes what it's doing well

File: Program.cs [hunk 8]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+
+      int index = 0;
+      foreach (int number in numbers)
+      {
+        if (index == 0)
+        {
+          // Skip this one.
+        }
```
Comment (SChandarana) [inline Program.cs:77]: Since you're using an index here you can replace the foreach loop with a for and then at replace `number` with `numbers[index]`. Also if you start the index at 1 then you don't need to worry about the `skip this step` part

File: Program.cs [hunk 9]
```diff
@@ -5,11 +5,91 @@ namespace Calculator
...
+        {
+          if (op == "/")
+            answer = answer / number;
+          else if (op == "+")
+            answer = answer + number;
+          else if (op == "-")
+            answer = answer - number;
+          else
+            Console.WriteLine("Invalid operator");
+        }
```
Comment (SChandarana) [inline Program.cs:88]: This is missing * as an operator. The invalid operator check can also be removed if we move the validation to `AskForOperator()`
We can also change this to a switch statement to avoid needing to write lots of `else if`s
Does that make sense? :D

File: Calculator.sln [hunk 10]
```diff
@@ -15,6 +17,10 @@ Global
 		{42B5D5A2-97DE-41F7-A20D-FD68240F701E}.Debug|Any CPU.Build.0 = Debug|Any CPU
 		{42B5D5A2-97DE-41F7-A20D-FD68240F701E}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{42B5D5A2-97DE-41F7-A20D-FD68240F701E}.Release|Any CPU.Build.0 = Release|Any CPU
+		{1CB150D8-39B5-46D5-B16A-42B97E05C7AC}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{1CB150D8-39B5-46D5-B16A-42B97E05C7AC}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{1CB150D8-39B5-46D5-B16A-42B97E05C7AC}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{1CB150D8-39B5-46D5-B16A-42B97E05C7AC}.Release|Any CPU.Build.0 = Release|Any CPU
```
Comment (SChandarana) [inline Calculator.sln:23]: There are some inconsistencies with indent sizes through the different files. Here it looks like you're using tabs, in other files you're using 4 spaces and in another you're using 2 spaces. You can change your default tab spacing within your editor. For this project I think I prefer 4 spaces for an indent so if you could change this file and `Program.cs` to be consistent :D

File: Program.cs (hunk 4 above)
Comment (SChandarana) [reply Program.cs:27]: Also since we know what these values are going be you can change them from `var` to e.g. `int[]`

**Review by SChandarana — COMMENTED** [review]
Looks good, just a couple things to have a look at :D Let me know if anything doesn't make sense!
